### PR TITLE
samples: nrf9160: modem_shell: Use new nrf_modem_at API

### PR DESCRIPTION
**Pull request by trantanen**
Run AT commands using nrf_modem_at API instead of at_cmd library.
Notifications are handled via at_monitor library.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/link/link_shell_pdn.c [hunk 1]
```diff
@@ -337,25 +337,30 @@ int link_shell_pdn_activate(int pdn_cid)
 	return ret;
 }
 
-void link_shell_pdn_init(void)
+void link_shell_pdn_init_callbacks(void)
```
Comment (tokangas) [inline samples/nrf9160/modem_shell/src/link/link_shell_pdn.c:340]: *nit* Why the function name uses a plural, when it only sets one callback? Would something like `link_shell_pdn_events_subscribe()` be more descriptive?

Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/link/link_shell_pdn.c:340]: Yes. Changed.

File: samples/nrf9160/modem_shell/src/link/link.c [hunk 2]
```diff
@@ -640,19 +629,16 @@ void link_rai_read(void)
...
-	if (state == AT_CMD_OK) {
+	err = nrf_modem_at_printf("AT%%RAI=%d", enable);
+	if (!err) {
 		mosh_print(
 			"Release Assistance Indication functionality set to enabled=%d.\n"
 			"The change will be applied when going to normal mode for the next time.",
 			enable);
 	} else {
-		mosh_error("Error state=%d, error=%d", state, err);
+		mosh_error("Error=%d", err);
```
[the remark this answers is not in the document]
Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/link/link.c:641]: Changed

File: samples/nrf9160/modem_shell/src/link/link.c [hunk 3]
```diff
@@ -415,14 +413,14 @@ static int link_default_pdp_context_auth_set(void)
...
-			"Release 14 features AT-command \"%s\" returned: ERROR",
-			rel14feat_at_cmd);
+	int ret;
+
+	ret = nrf_modem_at_printf("AT%%REL14FEAT=1,1,1,1,1");
+	if (ret < 0) {
+		mosh_warn("Release 14 features AT-command failed, err %d", ret);
+	} else if (ret > 0) {
+		mosh_warn("Release 14 features AT-command error, type %d err %d",
+			nrf_modem_at_err_type(ret), ret);
```
[the remark this answers is not in the document]
Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/link/link.c:423]: Yes. Changed

File: samples/nrf9160/modem_shell/src/link/link_api.c [hunk 4]
```diff
@@ -386,194 +353,116 @@ void link_api_coneval_read_for_shell(void)
...
-#define AT_CMD_XMONITOR_RESP_BAND_INDEX 7
-#define AT_CMD_XMONITOR_RESP_CELL_ID_INDEX 8
-#define AT_CMD_XMONITOR_RESP_RSRP_INDEX 11
-#define AT_CMD_XMONITOR_RESP_SNR_INDEX 12
-
-#define AT_CMD_X_MONITOR_MAX_HANDLED_INDEX AT_CMD_XMONITOR_RESP_SNR_INDEX
-#define AT_CMD_XMONITOR_RESP_MAX_STR_LEN OPERATOR_FULL_NAME_STR_MAX_LEN
+/* For having a numeric constant in scanf string length */
+#define L_(x) #x
+#define L(x) L_(x)
```
[the remark this answers is not in the document]
Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/link/link_api.c]: Changed.

**Review by tokangas — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/link/link_api.c [hunk 5]
```diff
@@ -386,194 +353,115 @@ void link_api_coneval_read_for_shell(void)
...
-#define XMONITOR_RESP_CELL_ID_VALID 32
-#define XMONITOR_RESP_RSRP_VALID 64
-#define XMONITOR_RESP_SNR_VALID 128
-
-#define OPERATOR_FULL_NAME_STR_MAX_LEN 124
-#define OPERATOR_SHORT_NAME_STR_MAX_LEN 64
-#define OPERATOR_CELL_ID_STR_MAX_LEN 32
-#define OPERATOR_PLMN_STR_MAX_LEN 32
+
+#define OP_FULL_NAME_STR_MAX_LEN 128
```
Comment (hiltunent) [inline samples/nrf9160/modem_shell/src/link/link_api.c:357]: This has increased. Was it on purpose?

**Review by hiltunent — APPROVED** [review]

File: samples/nrf9160/modem_shell/src/link/link_api.c (hunk 5 above)
Comment (jhirsi) [reply samples/nrf9160/modem_shell/src/link/link_api.c:357]: yes, increased by 4 bytes. 128 is used elsewhere as well for this. Anyways, it is safe even though if modem would return even larger.

**Pull request merged**